iocage_lib/ioc_json.py

```diff
@@ -678,31 +681,29 @@ def get_version():
     def get_pool_and_iocroot(self):
         """For internal getting of pool and iocroot."""
         def get_pool():
+            # This function does following (keeping old behavior):
+            # 1) Ensures multiple activated pools aren't present
+            # 2) Activates first pool it finds until activate command has been
+            #  issued already ( keeping old behavior )
+            # 3) Only activate if pool is not freenas-boot and iocage skip
+            # is false
             old = False
-            zpools = list(map(lambda x: x.name, list(self.zfs.pools)))
-
-            match = 0
-
-            for pool in zpools:
-                prop_ioc_active = self.zfs_get_property(
-                    pool, "org.freebsd.ioc:active")
-                prop_comment = self.zfs_get_property(pool, "comment")
-
-                if prop_ioc_active == "yes":
-                    _dataset = pool
-                    match += 1
-                elif prop_comment == "iocage":
-                    _dataset = pool
-                    match += 1
+            matches = []
+            for pool in PoolListableResource():
+                ds = Dataset(pool.name)
+                if pool.active:
+                    matches.append(pool)
+                elif ds.properties['comment'] == 'iocage':
```

Comment:
Isnt comment a user property? Doesnt this need special handling? e.g. org.iocage:comment ? or does it already do that

Reply:
It indeed is but keeping the old implementation, it has not followed that convention. it just is `comment`

Reply:
I dont follow. `comment` is not a valid zfs property, how can you expect it to exist?

Reply:
I don't. I am just keeping the old behavior. Perhaps @skarekrow can elaborate ?

Reply:
It was used for legacy iocage installations. The property is a *zpool* property.

Reply:
Well in that case why are we checking it for zfs root dataset ?

Reply:
I don't remember the reason why, it was years ago :P

Reply:
Lets remove it then, it has been broken for years :D